### PR TITLE
[WIP] Complete refactoring

### PR DESCRIPTION
**Pull request by nathanaelhoun**
## Changes

Related to #4 

- [ ] Refactor commands 
	- [x] Move them in their own package `commands`
	- [ ] Adapt them so they work
	- [ ] Use handlers instead of a switch
- [ ] Refactor HTTP
	- [ ] Move it into its own package (??)
	- [ ] Use router (`mux.Router` ?) to route theme
	- [ ] Adapt it so it works
- [ ] Refactor plugin
	- [ ] Move it into its own package `plugin`
	- [ ] Adapt it to it works

- [ ] Update README.md (?)

## Changelog

```
## New features
- /

## Enhancements
- Complete code refactoring for a more robust and testable code.

## Bug Fixes and Others
- /
```

### CODE REVIEW

File: server/commands/commands.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package commands
```
Comment (darkLord19) [inline server/commands/commands.go:1]: i think commands.go should live in server/ directory and commands should contain only code files for individual commands.

Comment (nathanaelhoun) [reply server/commands/commands.go:1]: Fair point, will change this!

**Comment by nathanaelhoun** [comment]
@darkLord19 please do not force-push as it mess up the history when working on the same branch

**Comment by darkLord19** [comment]
> @darkLord19 please do not force-push as it mess up the history when working on the same branch

+1. was rebasing on top of master hence force push.

**Comment by nathanaelhoun** [comment]
Closing this, moving this work to #8.

**Pull request closed without merging**